tests/Type/data/paginator-extension.php

```diff
@@ -7,6 +7,19 @@
 use App\User;
 use function PHPStan\Testing\assertType;
 
-assertType('array', User::paginate()->all());
-assertType('array', User::simplePaginate()->all());
-assertType('array', User::cursorPaginate()->all());
```

Comment:
Please don't remove these tests.

Reply:
I would rather not remove tests, but, `$paginator->all()` is not the recommended way of retrieving the items from a paginator instance. The tests will also fail, because PHPStan is unable to figure our the right return type for `all()`, because it is inherited from the `Collection` @mixin, rather than a dedicated method on the Paginator instance.

https://laravel.com/docs/9.x/pagination#paginator-instance-methods

Reply:
It shouldn't matter if it is the recommended way or not. Like you said `AbstractPaginator` have `@mixin Collection` so it should be possible to do these collection calls on paginators.

> PHPStan is unable to figure our the right return type for all(), because it is inherited from the Collection @mixin
Why is that? PHPStan understands `@mixin`.

Reply:
I reported it to PHPStan in https://github.com/phpstan/phpstan/issues/8008, and it looks like to be fixed, though unreleased.

Reply:
Pardon me, it's already released so we can add the tests back. :)

Reply:
Done! ✅